proposals/esm-integration/README.md

```diff
@@ -88,57 +88,74 @@ Three things happen for each module during the Construction phase.
+For WebAssembly, evaluation consists of:
+- validating the WebAssembly module
+- snapshotting all imports:
+    - If any import is in TDZ, throw a ReferenceError exception
+    - Type-check/coerce each imported value against the declared import type, per step 8 of the [*instantiate a WebAssembly module*](https://webassembly.github.io/spec/js-api/index.html#instantiate-a-webassembly-module) algorithm.
+    - Initialize ("snapshot") the imported bindings to the checked/coerced value
+- initializing the Wasm module, performing [Wasm module instantiation](https://webassembly.github.io/spec/core/appendix/embedding.html#embed-instantiate-module)
+    - filling in memory using the data segment
+    - filling in tables using elem segments
+    - running the start function, which corresponds to running the top-level module code
+- initializing all exports in the Lexical Environment to their analogous Wasm/JS API objects, per step 4 of the [*instantiate a WebAssembly module*](https://webassembly.github.io/spec/js-api/index.html#instantiate-a-webassembly-module) algorithm.
 
-### Evaluation
+Because JS modules that the WebAssembly module imports from are already evaluated at this point, their values will be available for WebAssembly to snapshot (as long as the WebAssembly module was not in a cycle with the JS module, with the WebAssembly module's evaluation preceding the JS module's evaluation).
 
-During evaluation, the code is evaluated to assign values to the exported bindings. In JS, this means running the top-level module code.
+#### A note on snapshotting
 
-For WebAssembly, evaluation consists of:
-- filling in memory using the data segment
-- filling in tables using elem segments
-- running the start function, which corresponds to running the top-level module code
+For JavaScript ES modules, an export name is bound to a slot on the Module Record. For all of the values that we currently allow, this slot will contain a reference to another memory location, which contains the object (e.g. the `WebAssembly.Global` object that is being imported).
+
+The term "snapshot" means that WebAssembly will snapshot the reference. This does result in an observable difference between the way JavaScript and WebAssembly handle updates. In JavaScript modules, it is possible for the exporting module to update an export to point to a different object. Other JavaScript modules that are importing that export will then get the new object. In contrast, while WebAssembly modules will see changes to the object that the export was first bound to (e.g. when the exporting module calls `WebAssembly.Global.prototype.set` to change the value), the WebAssembly module will never see updated bindings.
```

Comment:
I don't think WebAssembly.Global is a good example here, because it could be confusing when using mutable globals.

Reply:
I don't quite understand--the point of this paragraph is to work through that confusion and explain what the semantics are.

Reply:
Ok now I understand, nevermind